test/libsolidity/SemanticTest.cpp

```diff
@@ -149,11 +180,28 @@ TestCase::TestResult SemanticTest::runTest(ostream& _stream, string const& _line
 
 	bool constructed = false;
 
+	// Iterate through the test calls and set the previous call.
+	TestFunctionCall* previousCall{nullptr};
 	for (auto& test: m_tests)
 	{
+		test.setPreviousCall(previousCall);
+		test.setTestHooks(&m_testHooks);
+		previousCall = &test;
+	}
+
+	for (auto& hook: m_testHooks)
+		hook->beginTestCase();
+
+	for (auto& test: m_tests)
+	{
+		for (auto& hook: m_testHooks)
+			hook->beforeFunctionCall(test);
+
 		if (constructed)
 		{
-			soltestAssert(test.call().kind != FunctionCall::Kind::Library, "Libraries have to be deployed before any other call.");
+			soltestAssert(
+				test.call().kind != FunctionCall::Kind::Library,
+				"Libraries have to be deployed before any other call.");
 			soltestAssert(
 				test.call().kind != FunctionCall::Kind::Constructor,
 				"Constructor has to be the first function call expect for library deployments.");
```

Comment:
I know it is old code, but maybe applying it as we are touching this part of code?
```suggestion
			soltestAssert(
				test.call().kind != FunctionCall::Kind::Constructor,
				"Constructor has to be the first function call expect for library deployments."
			);
```